### DISCUSSION TITLE
Spatially-dependent timestep size

### DISCUSSION THREAD

**Discussion by WilkAndy** [idaholab/moose#20320] [Q&A General] [answered]
I'm currently in a meeting where someone mentioned their software could use spatially-dependent timestep size.  Eg, if nothing-much is happening in a certain domain, then why not take large timesteps there?  Has anyone thought about doing this with MOOSE?

andy

**Comment by GiudGiud** [comment, marked as answer]
Hello

For a per-domain granularity:
Seems like something we ll be able to do when we support multi-system. First we are reworking the Executioner system to allow arbitrary chaining (see the Executor system), then we will look at multi-system.

For a per-element granularity of the time & timestep,
I dont think that s planned. Could be interesting but seems like a headache to keep track of time everywhere and always be interpolating.

Guillaume
